packages/exchange/tests/providers/conftest.py

```diff
@@ -128,4 +149,4 @@ def vision(provider_cls: Type[Provider], model: str, **kwargs) -> Tuple[Message,
             content=[ToolResult(tool_use_id="xyz", output='"image:tests/test_image.png"')],
         ),
     ]
-    return provider.complete(model=model, system=system, messages=messages, tools=None, **kwargs)
+    return provider.complete(model=model, system=system, messages=messages, tools=(), **kwargs)
```

Comment:
There are a lot of IDE warnings about call sites we use, as it is at the moment `tools: Tuple[Tool]`. So, this clears warnings about passing None. Though I wonder if it shouldn't be `tools: Tuple[Tool, ...]`? Seems that would be more explicit about possibly none, and also clear some other warnings. If you feel this is worthwhile I can in a follow-up or different PR.

<img width="795" alt="Screenshot 2024-10-05 at 7 32 24 AM" src="https://github.com/user-attachments/assets/f862d74e-4a7c-4139-ba08-1110c668ec18">

Reply:
https://github.com/block-open-source/goose/pull/149